Don't require libc++. macports-libstdc++ handles C++11 just fine too.
Don't put -std=c++11 in CPPFLAGS. That's not where it belongs.
--- configure.orig	2019-01-01 19:21:30.000000000 -0600
+++ configure	2019-01-10 21:25:59.000000000 -0600
@@ -17807,8 +17807,7 @@
 case $build_vendor in
 	apple)
 		EXEMPI_PLATFORM_DEF=MAC_ENV
-		LDFLAGS="$LDFLAGS -framework CoreServices -stdlib=libc++"
-		CPPFLAGS="$CPPFLAGS -std=c++11 -stdlib=libc++"
+		LDFLAGS="$LDFLAGS -framework CoreServices"
 		;;
 	*)
 		EXEMPI_PLATFORM_DEF=UNIX_ENV
